dbt_copilot_helper/commands/dns.py

```diff
@@ -251,42 +242,49 @@ def create_hosted_zones(client, base_domain, subdomain):
     zone_ids = {}
 
     for domain_name in domains_to_create:
+        domain_zone = f"{domain_name}."
         parent_domain = domain_name.split(".", maxsplit=1)[1]
         parent_zone = f"{parent_domain}."
-        parent_zone_id = _get_zone_id_or_abort(client, parent_zone)
+        hosted_zone_response = client.list_hosted_zones_by_name()
+        hosted_zones = {hz["Name"]: hz for hz in hosted_zone_response["HostedZones"]}
+        parent_zone_id = _get_zone_id_or_abort(hosted_zones, parent_zone)
 
-        if not click.confirm(
-            click.style("About to create domain: ", fg="cyan")
-            + click.style(f"{domain_name}\n", fg="white", bold=True)
-            + click.style("Do you want to continue?", fg="cyan"),
-        ):
-            exit()
-
-        click.secho(f"Creating hosted zone for {domain_name}...", fg="yellow")
+        if domain_zone in hosted_zones:
+            click.secho(f"Hosted zone '{domain_zone}' already exists", fg="yellow")
+            zone_ids[domain_name] = hosted_zones[domain_zone]["Id"]
```

Comment:
I'm assuming here that if we do not need to create the hosted zone because it already exists then we also do not need to copy the records to and from its parent.